src/fastga_he/gui/power_train_network_viewer.py

```diff
@@ -57,65 +187,437 @@ def power_train_network_viewer(
+    node_image_sequences = {}
+    propeller_rotation_sequences = {}
+
+    for component_name, component_type, om_type, icon_name, icon_size in zip(
+        names, components_type, components_om_type, icons_name, icons_size
     ):
-        if "propulsor" in component_type:
-            level = 0
-        elif ("propulsive_load" in component_type) and (component_name in prop_loads):
-            level = 1
-        elif "source" in component_type:
-            if component_name in prop_loads:
-                level = 1
-            else:
-                level = distance_from_prop_loads[component_name] + 1
-        else:
-            level = distance_from_prop_loads[component_name] + 1
-
-        graph.add_node(
-            component_name,
-            value=icon_size,
-            level=level,
-            shape="image",
-            image="file://" + icons_dict[icon_name],
-        )
+        graph.add_node(component_name)
+        node_sizes[component_name] = icon_size
+        node_types[component_name] = component_type
+        node_om_types[component_name] = om_type
+        node_icons[component_name] = icon_name
 
+    # Add edges
     for connection in connections:
-        # When the component is connected to a bus, the output number is also specified but it
-        # isn't meaningful when drawing a graph, so we will just filter it
-        if type(connection[0]) is list:
-            source = connection[0][0]
-        else:
-            source = connection[0]
+        # Filter out bus connection output numbers
+        source = connection[0][0] if isinstance(connection[0], list) else connection[0]
+        target = connection[1][0] if isinstance(connection[1], list) else connection[1]
+        graph.add_edge(source, target)
+
+    # Apply Graphviz layout algorithm
+    graph.layout(prog=layout_prog)
+
+    # Extract positions from Graphviz layout
+    position_dict = {}
+    for node in graph.nodes():
+        x, y = map(float, node.attr["pos"].split(","))
+        position_dict[node] = (x, y)
+
+    # Normalize positions for Bokeh
+    if position_dict:
+        x_coordinates = [coords[0] for coords in position_dict.values()]
+        y_coordinates = [coords[1] for coords in position_dict.values()]
+        x_min, x_max = min(x_coordinates), max(x_coordinates)
+        y_min, y_max = min(y_coordinates), max(y_coordinates)
+
+        x_range = x_max - x_min if x_max > x_min else 1
+        y_range = y_max - y_min if y_max > y_min else 1
+
+        # Scale to a reasonable display size with different orientation
+        if orientation == "TB" or orientation == "BT":
+            x_factor = 0.5
+            y_factor = 1.0
+            scale = 550
+            plot_width = 1200
+            plot_height = 900
+            x_range_max = 600
+            y_range_max = 600
+            icon_factor = 1
+            icon_width_factor = 1.0
+            x_orientation_offset = 125
+            y_orientation_offset = 0.0
```

Comment:
Some of these (maybe not all) should be input. For instance the plt height and width. For any other that you judge worth having as input, you could also think about having a default based on the orientation
+ For the sake of the maintanibility of the code, I wouldn't min a few comment about what those parameters are.